src/main/java/de/thetaphi/forbiddenapis/maven/SignaturesArtifact.java

```diff
@@ -51,10 +51,10 @@ public final class SignaturesArtifact {
   public String path;
   
   /** Used by the mojo to fetch the artifact */
-  Artifact createArtifact(ArtifactFactory artifactFactory) {
+  Artifact createArtifact() {
     if (groupId == null || artifactId == null || version == null || type == null) {
       throw new NullPointerException("signaturesArtifact is missing some properties. Required are: groupId, artifactId, version, type");
     }
-    return artifactFactory.createArtifactWithClassifier(groupId, artifactId, version, type, classifier);
+    return new DefaultArtifact(groupId, artifactId, classifier, type, version);
```

Comment:
as long as type = extension no additional mapper needed, otherwise you need to inject https://maven.apache.org/ref/3.5.0/maven-core/apidocs/org/apache/maven/artifact/handler/manager/ArtifactHandlerManager.html and retrieve the ArtifactHandler for the given type to resolve its extension.